06-building-a-program-start-to-finish/writing-tests/output/spec/InvertedIndex.spec.js

```diff
@@ -0,0 +1,211 @@
+      },
+
+      {
+        title: 'The Lord of the Rings: The Fellowship of the Ring.',
+        text: `An unusual alliance of man, elf, dwarf, 
+          wizard and hobbit seek to destroy a powerful ring.`
+      },
+      {
+        title: 'The Lord of the Rings: The Fellowship of the Ring.',
+        text: `An unusual alliance of man, elf, dwarf, 
+          wizard and hobbit seek to destroy a powerful ring.`
+      }
+    ];
+  });
+  describe('InvertedIndex class', () => {
+    it('should check that the class has a createIndex method', () => {
+      expect(typeof indexInstance.createIndex).toBe('function');
+    });
+
+    it('should check that the class has a readFile method', () => {
+      expect(typeof InvertedIndex.readFile).toBe('function');
+    });
+
+    it('should check that the class has a validateFile method', () => {
+      expect(typeof InvertedIndex.validateFile).toBe('function');
+    });
+
+    it('should check that the class has a tokenize method', () => {
+      expect(typeof InvertedIndex.tokenize).toBe('function');
+    });
+
+    it('should check that the class has a getDocumentTokens method', () => {
+      expect(typeof InvertedIndex.getDocumentTokens).toBe('function');
+    });
+
+    it('should check that the class has a getIndex method', () => {
+      expect(typeof indexInstance.getIndex).toBe('function');
+    });
+
+    it('should check that the class has a searchIndex method', () => {
+      expect(typeof indexInstance.searchIndex).toBe('function');
+    });
+
+    it('should check that the class has a getSearchResults method', () => {
+      expect(typeof indexInstance.getSearchResults).toBe('function');
+    });
+
+    it('should check that the class has a getDocuments method', () => {
+      expect(typeof indexInstance.getDocuments).toBe('function');
+    });
+  });
+
+  describe('Create Index', () => {
+    it('should return mapped indices to words in a JSON file', () => {
+      const expectedResult =
+        { alice: [0],
+          falls: [0],
+          into: [0],
+          a: [0, 1, 2],
+          rabbit: [0],
+          hole: [0],
+          and: [0, 1, 2],
+          enters: [0],
+          world: [0],
+          full: [0],
+          of: [0, 1, 2],
+          imagination: [0],
+          in: [0],
+          wonderland: [0],
+          an: [1, 2],
+          unusual: [1, 2],
+          alliance: [1, 2],
+          man: [1, 2],
+          elf: [1, 2],
+          dwarf: [1, 2],
+          wizard: [1, 2],
+          hobbit: [1, 2],
+          seek: [1, 2],
+          to: [1, 2],
+          destroy: [1, 2],
+          powerful: [1, 2],
+          ring: [1, 2],
+          the: [1, 2],
+          lord: [1, 2],
+          rings: [1, 2],
+          fellowship: [1, 2] };
+      indexInstance.createIndex(books, 'books');
+      expect(indexInstance.filesIndexed.books.index)
+        .toEqual(expectedResult);
+    });
+    it('should return false for incorrect document structure', () => {
```

Comment:
The reason for this is to still validate file even if the validateFile method is not called before trying to create index. It's just to double check.